server/routes/auth.ts

```diff
@@ -70,7 +70,7 @@ export async function reloadAuthentication() {
 	}
 
 	// These routes need to be redefined on every instance of a new router
-	router.post("/confirm", validateAndCacheHostName, postParser, async (request, response) => {
+	router.post("/confirm", authenticateWithReject, validateAndCacheHostName, postParser, async (request, response) => {
```

Comment:
Because this middleware is using `!request.user.verifiedEmail || !request.user.accountConfirmed` to check if a user is valid, a confirming user will always be rejected.

Reply:
Good catch, switched to explicitly validating